Services/Administration/OJS.Services.Administration.Data/Implementations/TestsDataService.cs

```diff
@@ -30,5 +30,10 @@ public async Task DeleteByProblem(int problemId)
             this.Delete(t => t.ProblemId == problemId);
             await this.SaveChanges();
         }
+
+        public async Task<int> GetProblemIdByTestId(int testId)
+            => await this.GetByIdQuery(testId)
+                .Select(t => t.ProblemId)
+                .FirstAsync();
```

Comment:
FirstAsync will throw exception if there is no test with the provided Id. In DataServices we tend to return null or default if the value is missing and let the caller decide what to do, instead of throwing exception.